Source/WebCore/PAL/pal/text/TextEncodingRegistry.cpp

```diff
@@ -254,8 +254,19 @@ std::unique_ptr<TextCodec> newTextCodec(const TextEncoding& encoding)
     Locker locker { encodingRegistryLock };
 
     ASSERT(textCodecMap);
+    if (!encoding.isValid()) {
+        WTFLogAlways("Trying to create new text codec with invalid (null) encoding name.");
+        return nullptr;
+    }
     auto result = textCodecMap->find(encoding.name());
-    ASSERT(result != textCodecMap->end());
+    if (result == textCodecMap->end()) {
+        WTFLogAlways("Can't find codec for valid encoding %s .", encoding.name());
+        return nullptr;
+    }
+    if (!result->value) {
```

Comment:
This whole if check does nothing. The ending return statement would do the same thing.

Reply:
Never mind this comment. This does something.